python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -640,7 +646,13 @@ def asset_partitions_with_newly_updated_parents_and_new_cursor(
                         f"Could not map partitions between parent {parent_asset_key.to_string()} "
                         f"and child {child_asset_key.to_string()}."
                     ) from e
-                for child_partition in child_partitions_subset.get_partition_keys():
+
+                child_partitions = reversed(list(child_partitions_subset.get_partition_keys()))
+                max_child_partitions = os.getenv("DAGSTER_MAX_AMP_CHILD_PARTITIONS", None)
```

Comment:
nit could move the int() here

Reply:
int(`None`) wouldn't work right? could default to 0, but seems a bit sketch to "default to 0 partitions", even if it works out ok in practice